sdk/core/Azure.Core/src/Cryptography/IKeyEncryptionKey.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core.Cryptography
+{
+    /// <summary>
+    /// A key which is used to encrypt, or wrap, another key
+    /// </summary>
+    public interface IKeyEncryptionKey
```

Comment:
Wouldn't a `IKeyEncryptor` be a better name?

Reply:
That's what I originally was going to call it.  However, when this is fed to storage it's referred to as keyEncryptionKey so I felt this would more explicitly link the two together.

Reply:
```IKeyEncryptionKey``` does look strange. I think storage should still call the parameter whatever it think is best, but the type name could be what we think is best. BTW, is the interface specific to wrapping and unwrapping key? Or would it work on any byte array? If the later, we should call it without reffering to "key" at all.

Reply:
This interface is specific to encrypting keys, in that it is only able to encrypt a single block as it doesn't use a nonce or IV.  Also the algorithm specified is expected to be a key wrap algorithm, not just any encryption algorithm. I personally think that IKeyEncryptionKey is an appropriate name as it's an abstraction of a key which is able to wrap another key.  While the name might seem odd without prior knowledge  of cryptography concepts, it is a well known term in cryptography and key hierarchy management. https://csrc.nist.gov/glossary/term/Key_Encryption_Key